### PR TITLE
[Windows] FlyoutPage: update CollapseStyle at runtime

### LINKED ISSUE #18200: Flyout Page SetCollapseStyle doesn't have any change
### Description

I'm trying to create a New MAUI app in windows wth Flyout Navigation.

And in Windows I want the Flyout should be open partially always and when I click on the Hamburger it should open fully.

### Steps to Reproduce

Create a MAUI App. 

Create a FlyoutPage and in the code behind add `On<Windows>().SetCollapseStyle(CollapseStyle.Partial);`

Expected:
image

Actual:

image

image

### Link to public reproduction project repository

_No response_

### Version with bug

7.0.96

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

7.0.92

### Affected platforms

Windows

### Affected platform versions

Windows 11

### Did you find any workaround?

No

### Relevant log output

```shell
NA
```

### PR DESCRIPTION
**Pull request by devanathan-vaithiyanathan**
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

### Issue details
FlyoutPage on Windows did not update its layout when the CollapseStyle property changed at runtime.

### Description of Change

This update enables dynamic support for the CollapseStyle property in FlyoutPage on Windows. It allows the flyout pane to update at runtime when the property changes,

### Issues Fixed

Fixes #18200 

**Tested the behavior in the following platforms.**
- [x] Windows

| Before  | After  |
|---------|--------|
| **Windows**<br> <video src="https://github.com/user-attachments/assets/9d7844d7-af65-465a-abb3-b611290afe1f" width="400" height="250"> |**Windows**<br> <video src="https://github.com/user-attachments/assets/2edd0934-c369-4dda-8269-e22291769c2e" width="400" height="250"> |

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# if TEST_FAILS_ON_ANDROID && TEST_FAILS_ON_CATALYST && TEST_FAILS_ON_IOS  // It's a Windows specific API issue, so restricting the other platforms
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs]: Because is a platform specific, here can use directly `#if WINDOWS`. Are not failing on other platforms, just have not sense.

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs (hunk 1 above)
Comment (devanathan-vaithiyanathan) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs]: @jsuarezruiz , Thanks for pointing out. I have modified the changes

**Comment by kubaflo** [comment]
## 🤖 AI Summary


<details>
<summary>📊 <strong>Expand Full Review</strong> — <a href="https://github.com/dotnet/maui/commit/a1ad2f86f1f32bd10996e61572d8b4916b08a601"><code>a1ad2f8</code></a> · <strong>condition compilation removed</strong></summary>

---

<details>
<summary><strong>🔍 Pre-Flight — Context & Validation</strong></summary>

---

**Issue:** #18200 - Flyout Page SetCollapseStyle doesn't have any change
**PR:** #29927 - [Windows] Fix Flyout Page SetCollapseStyle doesn't have any change
**Platforms Affected:** Windows
**Files Changed:** 2 implementation, 2 test

### Key Findings
- The linked issue reports that `WindowsSpecific.FlyoutPage.SetCollapseStyle(...)` has no runtime effect on Windows MAUI, although the same API worked in Xamarin.Forms.
- The PR updates the Windows platform configuration property to notify handlers on change and adds a Windows mapper to translate `CollapseStyle` into WinUI `NavigationViewPaneDisplayMode`.
- Test coverage was added as a HostApp issue page plus a shared UI test, so the validation path is a Windows UI test (`Issue18200`).
- Prior review feedback already questioned whether the mapper change stays consistent when flyout behavior updates later and whether the test truly proves the visual behavior difference between `Full` and `Partial` collapse styles.

### Edge Cases / Discussion Notes
- The issue thread includes a workaround that directly sets WinUI `PaneDisplayMode` through a mapper, which is conceptually close to the PR approach.
- Inline review feedback raised a possible interaction where `FlyoutBehavior` updates could overwrite the collapse-style-specific `PaneDisplayMode` after the property changes.
- The UI test is intentionally Windows-only because the API itself is Windows-specific.
- No prior PRAgent review comment was found to resume from.

### File Classification
- Implementation: `src/Controls/src/Core/FlyoutPage/FlyoutPage.Mapper.cs`
- Implementation: `src/Controls/src/Core/PlatformConfiguration/WindowsSpecific/FlyoutPage.cs`
- Test (HostApp UI page): `src/Controls/tests/TestCases.HostApp/Issues/Issue18200.cs`
- Test (Shared UITest): `src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs`
- Test type: UI test via HostApp + Appium shared test

### Fix Candidates
| # | Source | Approach | Test Result | Files Changed | Notes |
|---|--------|----------|-------------|---------------|-------|
| PR | PR #29927 | Trigger handler updates when `CollapseStyle` changes, then map Windows `CollapseStyle` values to WinUI `PaneDisplayMode` at runtime |  PENDING (Gate) | `src/Controls/src/Core/FlyoutPage/FlyoutPage.Mapper.cs`, `src/Controls/src/Core/PlatformConfiguration/WindowsSpecific/FlyoutPage.cs`, `src/Controls/tests/TestCases.HostApp/Issues/Issue18200.cs`, `src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs` | Original PR |

</details>

---

<details>
<summary><strong>🚦 Gate — Test Verification</strong></summary>

---

Result:  PASSED

### Gate Result:  PASSED

**Platform:** windows
**Mode:** Full Verification

- Tests FAIL without fix: 
- Tests PASS with fix: 

### Notes
- Verification used `Issue18200` on Windows.
- Without the fix, the UI test timed out waiting for the flyout item state change, reproducing the bug.
- With the fix applied, the same test passed, confirming the PR resolves the runtime collapse-style update issue.

</details>

---

<details>
<summary><strong>🔧 Fix — Analysis & Comparison</strong></summary>

---

### Fix Candidates
| # | Source | Approach | Test Result | Files Changed | Notes |
|---|--------|----------|-------------|---------------|-------|
| 1 | try-fix | Route CollapseStyle through `MauiNavigationView`'s flyout-behavior pipeline using a pane-display-mode override instead of mutating `PaneDisplayMode` directly in the mapper |  PASS | 3 files | Robust against later `FlyoutBehavior` updates overwriting the collapse style |
| 2 | try-fix | Use a virtual flyout-pane-display-mode hook in `MauiNavigationView`/`RootNavigationView` so `UpdatePaneDisplayModeFromFlyoutBehavior` natively resolves compact vs minimal mode |  PASS | 4 files | Template-method style alternative that keeps collapse-style behavior inside the existing pipeline |
| 3 | try-fix | Reuse existing `PinPaneDisplayModeTo` support on `RootNavigationView`, set it from the new CollapseStyle mapper, then re-run `IFlyoutView.FlyoutBehavior` mapping |  PASS | 2 files | Smallest passing alternative before cross-pollination; leans on existing navigation-view behavior |
| 4 | try-fix | Use a Windows resource flag on the navigation view and let `MauiNavigationView` re-evaluate flyout behavior against that flag when CollapseStyle changes |  PASS | 2 files | Passes, but relies on a string-keyed resource flag and removes an existing flyout-behavior cache guard |
| 5 | try-fix | Wrap the existing Windows `FlyoutBehavior` mapper with `ModifyMapping`, then apply `CollapseStyle` after the base mapper runs and re-trigger via `UpdateValue(FlyoutBehavior)` on property change |  PASS | 2 files | Simplest robust passing option; keeps the behavior in the existing mapper pipeline without adding native-view state |
| 6 | try-fix | Let Windows native flyout behavior pull `CollapseStyle` directly from the virtual view via an internal interface and dual cache on behavior + collapse mode |  PASS | 6 files | Works, but materially broader than needed and introduces new cross-assembly coupling |
| 7 | try-fix | Register a native `PaneDisplayModeProperty` callback to re-enforce the requested collapse style whenever internal/native logic overwrites it |  PASS | 2 files | Passed after retries, but callback enforcement and weak-table bookkeeping are more invasive than necessary |
| 8 | try-fix | Register a shared `MapPaneConfiguration` mapper for both `FlyoutLayoutBehavior` and `CollapseStyle` and resolve the final native mode from combined state |  PASS | 2 files | Good unified-mapper variant, but less clearly aligned with the existing `FlyoutBehavior` update path than Candidate #5 |
| PR | PR #29927 | Add a property-changed callback and map `CollapseStyle` directly to WinUI `PaneDisplayMode` in the Windows mapper |  PASSED (Gate) | 4 files | Original PR |

### Cross-Pollination
| Model | Round | New Ideas? | Details |
|-------|-------|------------|---------|
| claude-opus-4.6 | 2 | Yes | Make the existing Windows `FlyoutBehavior` mapping compute `PaneDisplayMode` from both current `FlyoutBehavior` and attached `CollapseStyle`. Tested as Candidate #5. |
| claude-sonnet-4.6 | 2 | Yes | Let native flyout behavior pull `CollapseStyle` from the virtual view as source-of-truth. Tested as Candidate #6. |
| gpt-5.3-codex | 2 | No | NO NEW IDEAS |
| gemini-3-pro-preview | 2 | No | NO NEW IDEAS |
| claude-opus-4.6 | 3 | No | NO NEW IDEAS |
| claude-sonnet-4.6 | 3 | No | NO NEW IDEAS |
| gpt-5.3-codex | 3 | No | NO NEW IDEAS |
| gemini-3-pro-preview | 3 | Yes | Re-enforce collapse style through a native `PaneDisplayModeProperty` callback. Tested as Candidate #7. |
| gpt-5.3-codex | 4 | Yes | Use one shared mapper entrypoint for pane configuration across `FlyoutLayoutBehavior` and `CollapseStyle`. Tested as Candidate #8. |

**Exhausted:** Yes
**Selected Fix:** Candidate #5  Simplest robust passing solution. It keeps collapse-style handling in the existing `FlyoutBehavior` mapper pipeline without introducing new native-view state, new interfaces, or callback enforcement.

</details>

---

<details>
<summary><strong>📋 Report — Final Recommendation</strong></summary>

---

Final Recommendation: REQUEST CHANGES

##  Final Recommendation: REQUEST CHANGES

### Phase Status
| Phase | Status | Notes |
|---|---|---|
| Pre-Flight |  COMPLETE | Windows FlyoutPage runtime CollapseStyle update issue with UI coverage in `Issue18200` |
| Gate |  PASSED | Windows full verification: test failed without fix and passed with fix |
| Try-Fix |  COMPLETE | 8 attempts, 8 passing alternatives, Candidate #5 selected |
| Report |  COMPLETE | |

### Summary
The PR fixes the reported Windows bug for the current UI test, so the submitted implementation is functionally valid. However, the try-fix phase found a better alternative: Candidate #5 keeps CollapseStyle handling inside the existing `FlyoutBehavior` mapping pipeline and achieves the same passing behavior with fewer moving parts and lower risk of future drift.

### Root Cause
The Windows FlyoutPage stack already has a native `FlyoutBehavior` pipeline that determines `PaneDisplayMode`. The bug occurs because runtime `CollapseStyle` changes never trigger that pipeline, so Windows stays on the original pane mode. The PR addresses that by directly setting `PaneDisplayMode` from a separate mapper, but that approach bypasses the existing flow that also manages pane mode and can create a second source of truth.

### Fix Quality
The PR's fix passes the new test, but it directly mutates native state from a dedicated CollapseStyle mapper instead of integrating with the existing `FlyoutBehavior` update path. Candidate #5 is stronger because it wraps the existing Windows `FlyoutBehavior` mapping with `ModifyMapping`, applies `CollapseStyle` after the base mapper runs, and reuses `UpdateValue(nameof(IFlyoutView.FlyoutBehavior))` on property change. That keeps all pane-mode decisions in one mapper pipeline and avoids adding extra native-view state, interfaces, or callback enforcement.

</details>

---

</details>



<details>
<summary>📋 <strong>Expand PR Finalization Review</strong></summary>

## PR #29927 Finalization Review

### Title & Description

**Title:** Needs improvement

- **Current:** `[Windows] Fix Flyout Page SetCollapseStyle doesn't have any change`
- **Recommended:** `[Windows] FlyoutPage: update CollapseStyle at runtime`

**Assessment:** The current title captures the issue but reads awkwardly and does not clearly describe the implemented behavior. The recommended title is more searchable and matches the actual fix.

**Description:** Good enough, but could be stronger

What works:
- Includes the required NOTE block
- Clearly states the Windows-specific runtime-update problem
- Matches the main implementation at a high level
- Includes the linked issue and tested platform

Suggested improvements:
- Add the root cause: `CollapseStyle` changed the attached property value, but the handler was never notified, so `RootNavigationView.PaneDisplayMode` stayed stale.
- Mention the actual implementation details: a property-changed callback was added on `CollapseStyleProperty`, and a Windows mapper now translates `CollapseStyle` to WinUI `PaneDisplayMode`.
- Mention the added UI regression test for issue `#18200`.

### Code Review Findings

####  Important issue

**CollapseStyle can be overwritten by later FlyoutBehavior updates**
- **Files:** `src/Controls/src/Core/FlyoutPage/FlyoutPage.Mapper.cs:39-56`, `src/Controls/src/Core/FlyoutPage/FlyoutPage.cs:328-359`, `src/Core/src/Platform/Windows/MauiNavigationView.cs:173-204`
- **Problem:** `MapCollapseStyle` sets `RootNavigationView.PaneDisplayMode` directly, but Windows FlyoutPage re-runs `Handler.UpdateValue(nameof(FlyoutBehavior))` on size and display-orientation changes. That flows into `UpdatePaneDisplayModeFromFlyoutBehavior()`, which unconditionally resets `FlyoutBehavior.Flyout` to `LeftMinimal`, so a previously selected `CollapseStyle.Partial` can be lost after resize/orientation changes.
- **Why it matters:** The PR fixes the immediate runtime toggle case, but the chosen pane mode is not sticky across later layout-driven updates.
- **Recommendation:** Integrate `CollapseStyle` into the WinUI pane-mode calculation path, or pin the pane display mode instead of setting it once ad hoc, so later `FlyoutBehavior` remaps do not undo `CollapseStyle.Partial`.

####  Suggestion

**The UI test does not cover the reset path above**
- **Files:** `src/Controls/tests/TestCases.HostApp/Issues/Issue18200.cs`, `src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue18200.cs`
- **Observation:** The added test verifies toggling `CollapseStyle` during a steady-state session, but it does not exercise the size/orientation path that triggers `FlyoutBehavior` remapping.
- **Recommendation:** Extend the test coverage to validate that `CollapseStyle.Partial` remains effective after a window resize or orientation/display-info change.

### Overall

The PR metadata mostly matches the implementation, but I would update the title for clarity. More importantly, I would not consider the implementation fully finalized until the Windows pane-mode reset issue is addressed or explicitly ruled out.

</details>

**Review by kubaflo — CHANGES_REQUESTED** [review]
Could you please review the AI's suggestions?

**Comment by devanathan-vaithiyanathan** [comment]
> Could you please review the AI's suggestions?

@kubaflo , I have addressed the AI suggestion.

**Review by kubaflo — APPROVED** [review]
